src/woodbury_pd_mat.jl

```diff
@@ -84,3 +84,7 @@ end
 function PDMats.unwhiten!(r::DenseVecOrMat, W::WoodburyPDMat{<:Real}, x::DenseVecOrMat)
     return unwhiten!(r, PDMat(Symmetric(Matrix(W))), x)
 end
+
+# NOTE: the parameterisation to scale up the Woodbury matrix is not unique. Here we
+# implement one way to scale it.
+*(a::WoodburyPDMat, c::Real) = WoodburyPDMat(a.A, a.D * c, a.S * c)
```

Comment:
Out of interest, why do we not need `*(c::Real, a::WoodburyPDMat)` here? 
It looks like that this is defined because of this in the tests `c * W == W * c`

Reply:
I'm not sure actually

Reply:
It's dispatching to `PDMats`: 

```
julia> @which c * W
*(c::T, a::AbstractPDMat) where T<:Real in PDMats at /Users/alexrobson/.julia/packages/PDMats/G0Prn/src/generics.jl:26

julia> @which W * c
*(a::WoodburyPDMat, c::Real) in PDMatsExtras at /Users/alexrobson/Projects/Squads/2021-01-AmberHTP/PDMatsExtras.jl/src/woodbury_pd_mat.jl:90
```

Reply:
Do we actually not need this MR?

Reply:
Oh, I think I see. `PDMats` in generics is dispatching the mirror to this one: https://github.com/JuliaStats/PDMats.jl/blob/master/src/generics.jl#L26

This still needs to be defined, iiuc, so all is good.

Reply:
good catch, so we'll still need to define `*(c::T, a::WoodburyPDMat)`, right? 

I thought there was a `*(c::T, a::AbstractPDMat) = *(a::AbstractPDMat, c::T)` defined somewhere, but seems like, no

Reply:
I don't actually know what the best practice is for this. It seems superfluous to define `*(c::Real, a::WoodburyPDMat)` which is the same as `*(c::Real, a::AbstractPDMat)`.

So I think we are ok as it is.

Reply:
I'm a bit confused now why things work or don't work.

Basically, without this PR, we'll get overflow error

```julia
julia> 2.0 * W
ERROR: StackOverflowError:
Stacktrace:
 [1] *(::WoodburyPDMat{Float64,Array{Float64,2},Diagonal{Float64,Array{Float64,1}},Diagonal{Float64,Array{Float64,1}}}, ::Float64) at /Users/wytbella/.julia/packages/PDMats/G0Prn/src/generics.jl:25 (repeats 79984 times)

julia> W * 2.0
ERROR: StackOverflowError:
Stacktrace:
 [1] *(::WoodburyPDMat{Float64,Array{Float64,2},Diagonal{Float64,Array{Float64,1}},Diagonal{Float64,Array{Float64,1}}}, ::Float64) at /Users/wytbella/.julia/packages/PDMats/G0Prn/src/generics.jl:25 (repeats 79984 times)
```

With this PR, both dispatches result a WoodburyPDMat.

```julia
julia> tmp = 2.0 * W
4×4 WoodburyPDMat{Float64,Array{Float64,2},Diagonal{Float64,Array{Float64,1}},Diagonal{Float64,Array{Float64,1}}}:
  4.08076    -0.0420901  -1.2894     2.46722
 -0.0420901   2.21476     0.294211  -1.29334
 -1.2894      0.294211    4.47231   -5.48715
  2.46722    -1.29334    -5.48715   19.3228

julia> tmp2 =  W * 2.0
4×4 WoodburyPDMat{Float64,Array{Float64,2},Diagonal{Float64,Array{Float64,1}},Diagonal{Float64,Array{Float64,1}}}:
  4.08076    -0.0420901  -1.2894     2.46722
 -0.0420901   2.21476     0.294211  -1.29334
 -1.2894      0.294211    4.47231   -5.48715
  2.46722    -1.29334    -5.48715   19.3228
```

What I don't quite get is how

```julia
julia> @which 2.0 * W
*(c::T, a::PDMats.AbstractPDMat) where T<:Real in PDMats at /Users/wytbella/.julia/packages/PDMats/G0Prn/src/generics.jl:26
```

could dispatch could generate a WoodburyPDMat with the correct parameterisation, rather than resulting in generic PDMat

Reply:
The generic definition `*(c::T, a::AbstractPDMat) where {T<:Real} = a * c`

propagates `a` type, meaning it is still `WoodburyPDMat` when it hits the `a * c` specialised dispatch